lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -64,16 +64,20 @@ def allocate_load_balancer(action_handler, lb_spec, lb_options, machine_specs)
       if lb_options[:security_group_id]
         security_group = ec2.security_groups[:security_group_id]
       elsif lb_options[:security_group_name]
-        security_group = ec2.security_groups.filter('group-name', lb_options[:security_group_name])
+        security_group = ec2.security_groups.filter('group-name', lb_options[:security_group_name]).first
```

Comment:
Ah, I think I wrongly diagnosed the bug. The bug is not so much in the calling to the AWS API, but one of the log writes assumes a singular result. Thus this should work with multiple security groups, but fails. Will fix, brb.